test/e2e/set/scyllacluster/scyllacluster_auth.go

```diff
@@ -0,0 +1,163 @@
+		})
+
+		framework.By("Specifying auth token in agent config")
+
+		agentConfig := struct {
+			AuthToken string `yaml:"auth_token"`
+		}{
+			AuthToken: "42",
+		}
+		agentConfigData, err := yaml.Marshal(agentConfig)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		agentConfigSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "agent-config",
+				Namespace: f.Namespace(),
+			},
+			Type: corev1.SecretTypeOpaque,
+			Data: map[string][]byte{
+				naming.ScyllaAgentConfigFileName: agentConfigData,
+			},
+		}
+
+		_, err = f.KubeClient().CoreV1().Secrets(f.Namespace()).Create(ctx, agentConfigSecret, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		sc.Spec.Datacenter.Racks[0].ScyllaAgentConfig = agentConfigSecret.Name
+		_, err = f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Update(ctx, sc, metav1.UpdateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// TODO: restart should be triggered by the Operator
+		framework.By("Initiating a rolling restart")
+
+		gen, err := restartRack(ctx, f.KubeClient().AppsV1(), sc.Spec.Datacenter.Racks[0], sc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the ScyllaCluster to pick up token change")
+
+		rackGenerationUpgraded := func(sc *scyllav1.ScyllaCluster) (bool, error) {
+			stss, err := getStatefulSetsForScyllaCluster(ctx, f.KubeClient().AppsV1(), sc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			return stss[sc.Spec.Datacenter.Racks[0].Name].Status.ObservedGeneration == gen, nil
+		}
+
+		waitCtx2, waitCtx2Cancel := contextForRollout(ctx, sc)
+		defer waitCtx2Cancel()
+		sc, err = waitForScyllaClusterState(waitCtx2, f.ScyllaClient().ScyllaV1(), sc.Namespace, sc.Name, scyllaClusterRolledOut, rackGenerationUpgraded)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Accepting requests authorized using token from user agent config")
+		checkAgentAuth(ctx, hosts, agentConfig.AuthToken, func(err error) {
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		framework.By("Changing auth token in agent config")
+
+		agentConfig.AuthToken = "666"
+		agentConfigData, err = yaml.Marshal(agentConfig)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		agentConfigSecret.Data[naming.ScyllaAgentConfigFileName] = agentConfigData
+		_, err = f.KubeClient().CoreV1().Secrets(f.Namespace()).Update(ctx, agentConfigSecret, metav1.UpdateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Initiating a rolling restart")
+
+		gen, err = restartRack(ctx, f.KubeClient().AppsV1(), sc.Spec.Datacenter.Racks[0], sc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the ScyllaCluster to pick up token change")
+
+		rackGenerationUpgraded = func(sc *scyllav1.ScyllaCluster) (bool, error) {
+			stss, err := getStatefulSetsForScyllaCluster(ctx, f.KubeClient().AppsV1(), sc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			return stss[sc.Spec.Datacenter.Racks[0].Name].Status.ObservedGeneration == gen, nil
```

Comment:
`observed generation != reconciled generation` (racy)

Reply:
wonder if we can trigger it through the scylla cluster, although the reconciliation there is minimalistic

the rollout condition for a statefulset is
```
sts.Status.ObservedGeneration >= sts.Generation &&
sts.Status.CurrentRevision == sts.Status.UpdateRevision &&
sts.Status.ReadyReplicas == *sts.Spec.Replicas
```